specs/agents/tracing-transaction-grouping.md

```diff
@@ -0,0 +1,37 @@
+## Transaction Grouping
+
+Even though agents should choose a transaction name that has a reasonable cardinality,
+they can't always guarantee that.
+For example,
+when the auto-instrumentation of a job scheduling framework sets the transaction name to the name of the instrumented job,
+the agent has no control over the job name itself.
+While usually the job name is expected to have low cardinality,
+users might set dynamic parts as part of the job name, such as a UUID.
+
+In order to give users an option to group transactions whose name contain dynamic parts that don't require code changes,
+agents MAY implement the following configuration option:
+
+### `transaction_name_groups` configuration
+
+With this option,
+you can group transaction names that contain dynamic parts with a wildcard expression.
+For example,
+the pattern `GET /user/*/cart` would consolidate transactions,
+such as `GET /users/42/cart`, into a single group.
+
+|                |                                                                                          |
+|----------------|------------------------------------------------------------------------------------------|
+| Type           | `List<`[`WildcardMatcher`](../../tests/agents/json-specs/wildcard_matcher_tests.json)`>` |
```

Comment:
I was also debating wether to make it possible to configure custom name for wildcard matcher, such as `GET /user/*/cart=show cart,/foo/*/bar=FooBar"`.

While it can be useful in some instances, it makes the configuration a little be more complex and verbose so I decided to only support a simple list of wildcard matchers where the resulting transaction name would be the literal matcher. In most cases, the wildcard matchers should also be nicely readable, unlike regex.

Another reason why regex is not supported is that the performance impact would be a lot higher.